### PR TITLE
[policyinsights] Fix next_link

### PR DESCRIPTION
**Pull request by msyyc**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

the former fix [link](https://github.com/Azure/azure-rest-api-specs/pull/14035) does not fix it entirely.

related issue: 
https://github.com/Azure/azure-sdk-for-python/issues/17280

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by msyyc** [comment]
> Does it require all codegen to upgrade `autorest.core`?

@lmazuel advise to upgrade to 3.4.5

**Comment by weidongxu-microsoft** [comment]
@ArcturusZhang @qiaozha This change requires codegen upgrade to `autorest.core` 3.4.5. Let me and Yuchao know your opinion whether we need to wait for you to upgrade codegen before merging it.

**Comment by weidongxu-microsoft** [comment]
PR for Java to upgrade https://github.com/Azure/autorest.java/pull/1104

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json
Comment (weidongxu-microsoft) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json:637]: @msyyc Do you know why `?Next paging op for policy events` is added after `{nextLink}`? Usually only `{nextLink}` is required.

Comment (msyyc) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json:637]: @weidongxu-microsoft I have test it locally and find it has no influence for generated code. So I guess it is just annotation.

**Comment by dw511214992** [comment]
@weidongxu-microsoft We haven't started to use track2 JS codegen, and we are still using track1 JS codegen. However, track1 JS codegen still uses autorest core V2.
Besides, I remember the autorest 3.4.5 has a bug: the generated codes is always the folder of swagger readme.md. I'm not sure whether it's a special bug in my computer. Do you encounter this issue?
Thanks

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json
Comment (pilor) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json:637]: @najagasi and @iscai-msft were the original committers of the x-ms-paths.  I imagine the path needed to be "unique" between policyStates.json and policyEvents.json to pass some validation but thats just a guess.  They can comment further

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json
Comment (pilor) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json:1167]: @iscai-msft had changed this back to {nextLink} in the original PR (https://github.com/Azure/azure-rest-api-specs/pull/14035/commits/cf7a5ea09c6f889942dedfd6c903b8b82d97457e).  Should check with them

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json
Comment (iscai-msft) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json:637]: @pilor you're correct, `x-ms-paths` needs these two to be unique. So we hack the uniqueness by adding "query parameters"

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json
Comment (iscai-msft) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json:1167]: @pilor I ended up having to revert that change. In `"paths"`, we have to have a leading backslash, so I had to move that `nextLink` declaration into `x-ms-paths`. To distinguish it from the other `nextLink` we have in `x-ms-paths`, I added the "query parameters"

**Comment by weidongxu-microsoft** [comment]
Java upgrade to 3.4.5 https://github.com/Azure/azure-rest-api-specs/pull/15165

**Comment by weidongxu-microsoft** [comment]
@dw511214992

Got it. Will check with Yuchao.
Usually v2 core is going to be upgrade as well (if this is the case of validation).

We had a daily smoke test and it seems the output folder is fine. https://github.com/Azure/autorest.java/blob/v4/fluent-tests/report.md
We do observe that 3.4.5 seems do a bit more validation than previous version. In particular, it reports missing `required` on discriminator, which was not found in previous version.
```
ERROR: Semantic violation: Discriminator must be a required property. (components > schemas > DataFlow)
  discriminator: { propertyName: �[32m'type'�[39m }
    - file:///home/runner/work/autorest.java/autorest.java/azure-rest-api-specs/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/DataFlow.json:24:3
  Error: Semantic validation failed. There was some errors
```

**Comment by msyyc** [comment]
> @weidongxu-microsoft Weidong Xu FTE We haven't started to use track2 JS codegen, and we are still using track1 JS codegen. However, track1 JS codegen still uses autorest core V2. So I think this PR cannot be merged if it needs js sdk.
> Besides, I remember the autorest 3.4.5 has a bug: the generated codes is always the folder of swagger readme.md. I'm not sure whether it's a special bug in my computer. Do you encounter this issue?
> Thanks

I have also met the problem in local environment that the generated codes is always the folder of swagger readme.md. It is not autorest bug but the usage problem. According to my experience:
(1) run `autorest --reset && autorest --latest` to clear cache
(2) check the cmd and and readme.js.md configuration
(3) run `autorest --version=3.4.5 --python --track2 --python-mode=update --use=@autorest/python@5.8.4 --use=@autorest/modelerfour@4.19.2 --python-sdks-folder=/_/azure-sdk-for-python/sdk/  /_/azure-rest-api-specs/specification/policyinsights/resource-manager/readme.md` again

**Comment by weidongxu-microsoft** [comment]
@msyyc 
An alternative is to have directive to remove the `{nextLink}` for the languages that does not support this.

**Comment by weidongxu-microsoft** [comment]
It might be easier to set condition as e.g. `!$(python)`

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang @qiaozha @dw511214992 This change requires codegen upgrade to use `autorest.core 3.4.5`. Let me and Yuchao know your opinion whether you would like us to wait for you to upgrade codegen before merging it.

The track 1 generator cannot change the version of core to 3.4.5

**Comment by weidongxu-microsoft** [comment]
@ArcturusZhang Yuchao is using directive to exclude them for js/go/etc.

**Comment by msyyc** [comment]
@ArcturusZhang Please re-add go again once the pipeline is fixed.
@dw511214992 I add directive in js configuration and you could remove them after migration to track2

File: specification/policyinsights/resource-manager/readme.go.md
Comment (ArcturusZhang) [inline specification/policyinsights/resource-manager/readme.go.md:78]: Please remove this

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

**Comment by ArcturusZhang** [comment]
We currently disable track 1 Go generation because directive alone cannot solve the generation failure.
We will add Go generation back once we enable track 2 generation for Go

**Comment by dw511214992** [comment]
> > @weidongxu-microsoft Weidong Xu FTE We haven't started to use track2 JS codegen, and we are still using track1 JS codegen. However, track1 JS codegen still uses autorest core V2. So I think this PR cannot be merged if it needs js sdk.
> > Besides, I remember the autorest 3.4.5 has a bug: the generated codes is always the folder of swagger readme.md. I'm not sure whether it's a special bug in my computer. Do you encounter this issue?
> > Thanks
> 
> @dw511214992 I have also met the problem in local environment that the generated codes is always the folder of swagger readme.md. It is not autorest bug but the usage problem. According to my experience:
> (1) run `autorest --reset && autorest --latest` to clear cache
> (2) check the cmd and and readme.js.md configuration
> (3) run `autorest --version=3.4.5 --python --track2 --python-mode=update --use=@autorest/python@5.8.4 --use=@autorest/modelerfour@4.19.2 --python-sdks-folder=/_/azure-sdk-for-python/sdk/ /_/azure-rest-api-specs/specification/policyinsights/resource-manager/readme.md` again

thanks @msyyc

**Review by RAY-316 — APPROVED** [review]

File: specification/policyinsights/resource-manager/readme.go.md
Comment (msyyc) [reply specification/policyinsights/resource-manager/readme.go.md]: done

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**
